### PR TITLE
Bugfix/leaking update samples

### PR DESCRIPTION
**Pull request by RubenAstudillo**
When using `taffybar`, I saw it leaking memory under heavy network loads. The graph produced was the following 
taffybar-linux-x86_64-before. It showed the problem was around the functions `parseDevNet` and `updateSamples`. This was also confirmed by the retainer graph profiling. 
taffybar-linux-x86_64-before3
 Adding `seq` on certain places i could reduce the memory consumption by those closures as shown on the following retainer profile graph.
taffybar-linux-x86_64
and on the cost center profile.
taffybar-linux-x86_64

### CODE REVIEW

**Comment by IvanMalison** [comment]
@RubenAstudillo Do you mind adding an entry to the changelog for this?

File: src/System/Taffybar/Information/Network.hs [hunk 1]
```diff
@@ -103,7 +103,7 @@ monitorNetworkInterfaces interval onUpdate = void $ do
         let speedInfo = map sampleToSpeeds samples
         onUpdate speedInfo
         return samples
-      doUpdate = MV.modifyMVar_ samplesVar ((>>= doOnUpdate) . updateSamples)
+      doUpdate = MV.modifyMVar_ samplesVar (doOnUpdate <=< updateSamples)
```
Comment (IvanMalison) [inline src/System/Taffybar/Information/Network.hs]: Is this change necessary? How does this impact anything?

Comment (RubenAstudillo) [reply src/System/Taffybar/Information/Network.hs]: Not really, I thought it was clearer, but I will roll it back!.

**Comment by RubenAstudillo** [comment]
> @RubenAstudillo Do you mind adding an entry to the changelog for this?

OK, should this be under a 3.2.2 header or still on the 3.2.1 header?

File: src/System/Taffybar/Information/Network.hs (hunk 1 above)
Comment (RubenAstudillo) [reply src/System/Taffybar/Information/Network.hs]: done on rebase!
